machine/machine_core/timeout.py

```diff
@@ -16,13 +16,19 @@
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
 import signal
+from types import FrameType, TracebackType
+from typing import TYPE_CHECKING, Optional, Type
+
+if TYPE_CHECKING:
+    import machine_core.ssh_connection as ssh_connection
```

Comment:
This is better spelled as `from machine_core import ssh_connection`.  I'm not a fan of the `TYPE_CHECKING` block, though.  This it the circular dependency thing, right?  I feel like we need to sort that out in a better way, possibly through the use of an interface...

Reply:
This is a circular dependency yes, I don't feel inclined to fix that right now. Could be done later sure, but honestly just having some basic typing in testlib for machines is the goal for me. Like:

![image](https://github.com/cockpit-project/bots/assets/67428/510b18e8-4eb2-44b8-8207-56a210a5d1a6)

Which hmm seems wrong!

Reply:
Gotta love Python :snake: